src/components/organisms/NavBarSchedule/NavBarSchedule.tsx

```diff
@@ -134,13 +139,32 @@ export const NavBarSchedule: FC<NavBarScheduleProps> = ({ isVisible }) => {
     };
   }, [relatedVenueEvents, userEventIds, selectedDayIndex, getEventLocation]);
 
+  const downloadCalendarPersonal = useCallback(() => {
+    const dayStart = addDays(startOfToday(), selectedDayIndex);
+    const allPersonalEvents: PersonalizedVenueEvent[] = relatedVenueEvents
+      .map(prepareForSchedule(dayStart, userEventIds, "calendar"))
+      .filter((event) => event.isSaved);
+
+    createCalendar(allPersonalEvents, `${PLATFORM_BRAND_NAME}_Personal`);
+  }, [relatedVenueEvents, userEventIds, selectedDayIndex]);
+
+  const downloadCalendarAll = useCallback(() => {
+    createCalendar(relatedVenueEvents, `${PLATFORM_BRAND_NAME}_Full`);
+  }, [relatedVenueEvents]);
```

Comment:
There will be some minimal changes required here based on other suggestions in this review to call the separated `createCalendar` and the proposed `downloadCalendar` functions.

Reply:
Renamed: `downloadAllEventsCalendar`
```suggestion
  const downloadAllEventsCalendar = useCallback(() => {
    createCalendar(relatedVenueEvents, `${PLATFORM_BRAND_NAME}_Full`);
  }, [relatedVenueEvents]);
```

Reply:
fixed in https://github.com/sparkletown/sparkle/pull/1423/commits/ec68e834fab42e9549a285d7a454e4c29e1f24e4